### PR TITLE
CMake: Add More Pip Helpers

### PR DESCRIPTION
**Pull request by ax3l**
Add and document more helpers to quickly and iteratively install our PICMI Python bindings with a one-liner from CMake:
(after one has _once_ run the configuration one-liner)
```console
cmake --build build --target pip_install -j 4
```

This should accelerate development cycles.

### CODE REVIEW

File: Docs/source/install/cmake.rst [hunk 1]
```diff
@@ -217,11 +278,10 @@ Environment Variable          Default & Values                             Descr
 ``PYWARPX_LIB_DIR``           *None*                                       If set, search for pre-built WarpX C++ libraries (see below)
 ============================= ============================================ ================================================================
 
+Note that we currently change the ``WARPX_MPI`` default intentionally to ``OFF``, to simplify a first install form source.
```
Comment (PhilMiller) [inline Docs/source/install/cmake.rst]: Typo:
```suggestion
Note that we currently change the ``WARPX_MPI`` default intentionally to ``OFF``, to simplify a first install from source.
```

File: Docs/source/install/cmake.rst [hunk 2]
```diff
@@ -173,29 +176,87 @@ Run
...
+   cmake --build build --target pip_install -j 4
+
+**That's it!**
+You can now :ref:`run a first 3D PICMI script <usage-picmi>` from our :ref:`examples <usage-examples>`.
+
+Developers could now change the WarpX source code and then call the build line again to refresh the Python installation.
+
+.. note::
+
+   These commands builds one ``-DWarpX_DIMS=...`` dimensionality (default: ``3``) at a time.
```
Comment (EZoni) [inline Docs/source/install/cmake.rst]: Typo:
```suggestion
   These commands build one ``-DWarpX_DIMS=...`` dimensionality (default: ``3``) at a time.
```

File: Docs/source/install/cmake.rst [hunk 3]
```diff
@@ -173,29 +176,87 @@ Run
...
+Using this workflow has the advantage that it can build and package up multiple libraries with varying ``WarpX_DIMS`` into one ``pywarpx`` package.
 
 Environment variables can be used to control the build step:
 
 ============================= ============================================ ================================================================
 Environment Variable          Default & Values                             Description
 ============================= ============================================ ================================================================
 ``WARPX_COMPUTE``             NOACC/**OMP**/CUDA/SYCL/HIP                  On-node, accelerated computing backend
-``WARPX_DIMS``                ``"2;3;RZ"``                                 Simulation dimensionalities (semicolon-separated list)
+``WARPX_DIMS``                ``"1;2;3;RZ"``                               Simulation dimensionalities (semicolon-separated list)
```
Comment (EZoni) [inline Docs/source/install/cmake.rst:259]: In the other table above we write simply 3/2/1/RZ, without the extra formatting `"..."`. Shall we make them uniform (meaning either both without formatting like the one above or both with formatting like the one here -  I personally find the one without formatting easier to read from within the table in this case)?

**Review by EZoni — APPROVED** [review]
Thank you, Axel! I'm not an expert on the CMake part, so I don't have valuable feedback on those code changes, but I went through the new docs and it looks great, thanks for improving that (P.S. I noticed only one small typo, see comment).

**Review by roelof-groenewald — APPROVED** [review]
Thanks!

File: Docs/source/install/cmake.rst (hunk 3 above)
Comment (ax3l) [reply Docs/source/install/cmake.rst:259]: Ah, but the one here is a literal default that contains literal `;`s.
Above it's individually allowed values, where lists are not permitted.

Have no better idea at the moment for this one :grin:

**Pull request merged**